aiogram/types/input_file.py

```diff
@@ -68,13 +76,15 @@ def from_file(
         """
         if filename is None:
             filename = os.path.basename(path)
-        with open(path, "rb") as f:
-            data = f.read()
-        return cls(data, filename=filename, chunk_size=chunk_size)
+        buffer = io.BytesIO()
+        with open(path, "rb") as fp:
+            while chunk := fp.read(chunk_size):
+                buffer.write(chunk)
+        return cls(buffer, filename=filename, chunk_size=chunk_size)
 
     async def read(self, bot: "Bot") -> AsyncGenerator[bytes, None]:
-        buffer = io.BytesIO(self.data)
-        while chunk := buffer.read(self.chunk_size):
+        self.buffer.seek(0)
+        while chunk := self.buffer.read(self.chunk_size):
```

Comment:
This change makes BufferedInputFile unsafe for concurrency because when you try to send the same InputFile from different coroutines, you're trying to read from the same BytesIO object.

Example:
1. Coroutine 1 read chunk 1
2. Coroutine 2 read chunk 2
3. Coroutine 1 read chunk 3

After that, both coroutines send the corrupted file.